tests/system_tests/wandb_backend_spy/spy.py

```diff
@@ -34,24 +32,61 @@ def freeze(self) -> Iterator[WandbBackendSnapshot]:
             finally:
                 snapshot._spy = None
 
-    def post_graphql(self, contents: bytes) -> fastapi.Response | None:
-        """Intercept a GraphQL request."""
-        return None
+    def post_graphql(
+        self,
+        request_raw: bytes,
+        response_raw: bytes,
+    ) -> None:
+        """Spy on a GraphQL request and response."""
+        request = json.loads(request_raw)
+
+        with self._lock:
+            query: str | None = request.get("query")
+            variables: dict[str, Any] | None = request.get("variables")
+            if query is None or variables is None:
+                return
+
+            print("GOT QUERY: ", query)
+            print("WITH VARIABLES: ", variables)
```

Comment:
remove the print? should we just have logs if we want to capture this info?

Reply:
removed!